org.eclipse.jdt.launching/launching/org/eclipse/jdt/internal/launching/StandardVMType.java

```diff
@@ -789,21 +765,11 @@ protected LibraryInfo parseLibraryInfo(IProcess process) {
 		return null;
 	}
 
+	private static final Pattern WINDOWS_LINE_SEPARATOR = Pattern.compile("\\\\"); //$NON-NLS-1$
+
 	protected String[] parsePaths(String paths) {
-		List<String> list = new ArrayList<>();
-		int pos = 0;
-		int index = paths.indexOf(File.pathSeparatorChar, pos);
-		while (index > 0) {
-			String path = paths.substring(pos, index);
-			list.add(path);
-			pos = index + 1;
-			index = paths.indexOf(File.pathSeparatorChar, pos);
-		}
-		String path = paths.substring(pos);
-		if (!path.equals("null")) { //$NON-NLS-1$
-			list.add(path);
-		}
-		return list.toArray(new String[list.size()]);
+		// Windows separator '\' is a RegEx meta-character, while the UNIX separator is not
+		return File.pathSeparatorChar == '/' ? paths.split(File.pathSeparator) : WINDOWS_LINE_SEPARATOR.split(paths);
```

Comment:
This broke JVM detection. `'/'` is wrong, it must be `:`

Reply:
Fixed via https://github.com/eclipse-jdt/eclipse.jdt.debug/pull/437

Reply:
Sorry for that, I somehow confused it with `File.separator`.
Thanks for the fix! It looks good.

Reply:
Nothing to sorry, I've reviewed and merged that so it is my fault.